### PR TITLE
Perlmutter makefile

### PR DESCRIPTION
**Pull request by jychoi-hpc**
Hi. I am working in StellFoundry with Michael Churchill. Recently we tried to build BEAMS3D on Perlmutter and made this makefile. Please let me know how it looks. Please feel free to give any comment/advice for further update.

### CODE REVIEW

File: SHARE/make_perlmutter.inc [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+
+#######################################################################
+#            Define Compiler Flags
+#######################################################################
+  FLAGS_R = 
+  FLAGS_D = -g -O0 -fexternal-blas -fcheck=all -fbacktrace -Wextra \
+   -Wtarget-lifetime -fbounds-check -ffpe-trap=zero -finit-real=snan \
+   -ffree-line-length-0 -dynamic -fallow-argument-mismatch 
+  LIBS    = 
+  FLAGS_R += -g -fbacktrace -dynamic -fallow-argument-mismatch -ffree-line-length-0
```
Comment (lazersos) [inline SHARE/make_perlmutter.inc]: Why not just explicitly set `FLAGS_R` at line 17?

File: SHARE/make_perlmutter.inc [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+#######################################################################
+  LNAG = F
+  NAG_LIB =
+
+#######################################################################
+#            NETCDF Options
+#######################################################################
+  LNETCDF = T
+  NETCDF_INC = 
+  NETCDF_LIB =
```
Comment (lazersos) [inline SHARE/make_perlmutter.inc]: You don't need to explicitly set the link or compile include lines?

File: SHARE/make_perlmutter.inc [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+  GENE_DIR = $(GENE_PATH)
+  LIB_GENE = libgene.a
+  GENE_LIB = $(GENE_DIR)/$(LIB_GENE) \
+             -L$(FFTW_DIR)/lib -lfftw3 -lfftw3f \
+             -L$(SLEPC_DIR)/lib -lslepc 
+
+#######################################################################
+#            COILOPT++ Options
+#######################################################################
+  LCOILOPT = T
```
Comment (lazersos) [inline SHARE/make_perlmutter.inc]: Are you really linking to COILOPT++?

File: SHARE/make_perlmutter.inc [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+  COILOPT_INC = -I$(COILOPT_PATH)
+  COILOPTPP_DIR = $(COILOPT_PATH)
+  LIB_COILOPTPP = libcoilopt++.a
+  COILOPT_LIB = $(COILOPT_PATH)/$(LIB_COILOPTPP) 
+
+#######################################################################
+#            TERPSICHORE Options
+#######################################################################
+  LTERPSICHORE = T
```
Comment (lazersos) [inline SHARE/make_perlmutter.inc]: Are you really linking to TERPSICHORE?

File: SHARE/make_perlmutter.inc [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+  TERPSICHORE_INC = -I$(TERPSICHORE_PATH)
+  TERPSICHORE_DIR = $(TERPSICHORE_PATH)
+  LIB_TERPSICHORE = libterpsichore.a
+  TERPSICHORE_LIB = $(TERPSICHORE_DIR)/$(LIB_TERPSICHORE)
+
+#######################################################################
+#            TRAVIS Options
+#######################################################################
+  LTRAVIS = T
```
Comment (lazersos) [inline SHARE/make_perlmutter.inc]: Are you really linking to TRAVIS?

**Review by lazersos — CHANGES_REQUESTED** [review]
@jychoi-hpc I just want to confirm a few items in your makefile then we can go ahead and merge.

File: SHARE/make_perlmutter.inc (hunk 1 above)
Comment (jychoi-hpc) [reply SHARE/make_perlmutter.inc]: It makes sense. I will fix to use "FLAGS_R" at line 17.

File: SHARE/make_perlmutter.inc (hunk 2 above)
Comment (jychoi-hpc) [reply SHARE/make_perlmutter.inc]: I think it is better to set NETCDF flags here. I will fix.

File: SHARE/make_perlmutter.inc (hunk 3 above)
Comment (jychoi-hpc) [reply SHARE/make_perlmutter.inc]: No

File: SHARE/make_perlmutter.inc (hunk 4 above)
Comment (jychoi-hpc) [reply SHARE/make_perlmutter.inc]: No

File: SHARE/make_perlmutter.inc (hunk 5 above)
Comment (jychoi-hpc) [reply SHARE/make_perlmutter.inc]: No. I will turn off.

**Comment by jychoi-hpc** [comment]
@lazersos Thank you for the feedback. I made some changes. Can you look again?

**Review by lazersos — APPROVED** [review]
Looks good to me now.

**Pull request merged**